web/Services/PropertyService.cs

```diff
@@ -113,13 +112,32 @@ public void UpdateProperty(int propertyId, PropertyViewModel updateModel, bool i
         dbModel.AcceptsWithoutGuarantor = updateModel.AcceptsWithoutGuarantor ?? dbModel.AcceptsWithoutGuarantor;
 
         dbModel.Rent = updateModel.Rent ?? dbModel.Rent;
-        dbModel.Availability = updateModel.Availability;
-        if (updateModel.Availability == PropertyDbModel.Avail_AvailableSoon)
+
+        dbModel.TotalUnits = updateModel.TotalUnits ?? dbModel.TotalUnits;
+        dbModel.OccupiedUnits = updateModel.OccupiedUnits ?? dbModel.OccupiedUnits;
+
+        // Occupied state takes precedence over attempted update
+        // If no update, fallback to current value as usual
+        dbModel.Availability = dbModel.OccupiedUnits == dbModel.TotalUnits
+            ? AvailabilityState.Occupied
+            : updateModel.Availability ?? dbModel.Availability;
+
+        if (dbModel.Availability == AvailabilityState.AvailableSoon)
         {
-            dbModel.AvailableFrom = updateModel.AvailableFrom;
+            if (updateModel.Availability == AvailabilityState.AvailableSoon)
+            {
+                // If update succeeds in making the property "available soon", then use its from date
+                dbModel.AvailableFrom = updateModel.AvailableFrom;
+            }
+            else
+            {
+                // Keep date the same if we're not updating it here
+                dbModel.AvailableFrom = dbModel.AvailableFrom;
+            }

```

Comment:
This doesn't do anything right?

Reply:
Oh yeah, think that's leftover from an old ternary expression